src/Workspaces/CSharp/Portable/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -265,18 +273,17 @@ public static bool CanRemoveParentheses(this ParenthesizedExpressionSyntax node,
 
         private static bool RemovalMayIntroduceInterpolationAmbiguity(ParenthesizedExpressionSyntax node)
         {
-            // First, find the parenting interpolation. If we find a parenthesize expression first,
-            // we can bail out early.
+            // First, find the parenting interpolation.
+            // If we find a parenthesize expression first, it doesn't mean we can bail, we still need to check : or ::
+            // For example: Func<bool, string> lambda = x => ($"{(x ? "foo" : "bar")}"), the outer parenthesis expression 
+            // will let us wrongly think we can remove the inner parenthesis expression.
```

Comment:
I think we can still bail early when a parenthesis is met *first*. Based on the original comment, it seems the issue is we keep iterating over the ancestors after encountering an interpolation. In the example you gave, we should have broken out the loop at `{(x ? "foo" : "bar")}`, so the enclosing `ParenthesizedExpression` should never be reached. Looks like someone has mistaken the `break` for the break of loop :)

Reply:
make sense, I will update this part in next commit.